x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/syncjob/action/ListConnectorSyncJobsAction.java

```diff
@@ -0,0 +1,172 @@
+        public static final ParseField CONNECTOR_ID_FIELD = new ParseField("connector_id");
+        private static final ParseField PAGE_PARAMS_FIELD = new ParseField("pageParams");
+        private final PageParams pageParams;
+        private final String connectorId;
+        private final ConnectorSyncStatus connectorSyncStatus;
+
+        public Request(StreamInput in) throws IOException {
+            super(in);
+            this.pageParams = new PageParams(in);
+            this.connectorId = in.readOptionalString();
+            this.connectorSyncStatus = in.readOptionalEnum(ConnectorSyncStatus.class);
+        }
+
+        public Request(PageParams pageParams, String connectorId, ConnectorSyncStatus connectorSyncStatus) {
+            this.pageParams = pageParams;
+            this.connectorId = connectorId;
+            this.connectorSyncStatus = connectorSyncStatus;
+        }
+
+        public PageParams getPageParams() {
+            return pageParams;
+        }
+
+        public String getConnectorId() {
+            return connectorId;
+        }
+
+        public ConnectorSyncStatus getConnectorSyncStatus() {
+            return connectorSyncStatus;
+        }
+
+        @Override
+        public ActionRequestValidationException validate() {
+            // Pagination validation is done as part of PageParams constructor
+            return null;
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            super.writeTo(out);
+            pageParams.writeTo(out);
+            out.writeOptionalString(connectorId);
+            out.writeOptionalEnum(connectorSyncStatus);
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+            Request request = (Request) o;
+            return Objects.equals(pageParams, request.pageParams)
+                && Objects.equals(connectorId, request.connectorId)
+                && connectorSyncStatus == request.connectorSyncStatus;
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(pageParams, connectorId, connectorSyncStatus);
+        }
+
+        private static final ConstructingObjectParser<ListConnectorSyncJobsAction.Request, String> PARSER = new ConstructingObjectParser<>(
+            "list_connector_sync_jobs_request",
+            p -> new ListConnectorSyncJobsAction.Request(
+                (PageParams) p[0],
+                (String) p[1],
+                p[2] != null ? ConnectorSyncStatus.fromString((String) p[2]) : null
+            )
+        );
+
+        static {
+            PARSER.declareObject(constructorArg(), (p, c) -> PageParams.fromXContent(p), PAGE_PARAMS_FIELD);
+            PARSER.declareString(optionalConstructorArg(), CONNECTOR_ID_FIELD);
+            PARSER.declareString(optionalConstructorArg(), ConnectorSyncJob.STATUS_FIELD);
```

Comment:
Another appraoch of dealing with nulls directly in parser:
```
 (p, c) -> p.currentToken() == XContentParser.Token.VALUE_NULL ? null : ConnectorSyncStatus.connectorSyncStatus(p.text()),
 ```
 sharing just for visibility. Good to decide on a single best approach in the future